### ISSUE TITLE
/sites/$site/media/new/ doesn't accept non-image files in the `media_urls` for sideloading.

### ISSUE THREAD

**Issue by georgestephanis** [Automattic/jetpack#3164]
In `WPCOM_JSON_API_Endpoint::handle_media_sideload()`

``` php
if ( ! file_is_displayable_image( $tmp ) ) {
    @unlink( $tmp );
    return false;
}
```

https://github.com/Automattic/jetpack/blob/5daf269bfa2040388710443e0725827891721a31/class.json-api-endpoints.php#L1903-L1906

The `file_is_displayable_image()` is a WordPress core function that rejects anything other than a gif, jpg, png, or bmp -- https://github.com/WordPress/WordPress/blob/7ad65267b61fb50e862433b4139cb63fd4fcfec2/wp-admin/includes/image.php#L445-L474

So any media other than images (videos, audio, etc) gets shorted out and is not accepted, if passed in via `media_urls` array parameter rather than being uploaded with the request.

https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/media/new/

Is this by design, or an oversight?

**Comment by georgestephanis** [comment]
Relevant changeset: https://github.com/Automattic/jetpack/commit/690d135a3f2f7e30633e0553e0c92be1436a7640 by @jkudish -- any thoughts?

**Comment by dbtlr** [comment]
So here is the $20,000 question, do we simply remove this check, or should we add some sort of file extension check that also allows for common video and audio file types?

Ideally, I'd like to do something that does a sanity check to also check that the video file is readable and not just a faked file extension, however this would need to wait until the transcoding step. So file-extension is probably the best bet for a quick MIME-type check.

I could probably make a future case for allowing people to also upload files like PDFs and DOCs, if they're providing something like a free ebook and we decide to provide some sort of cheap file hosting / delivery service that exists as part of premium jetpack. This could be stretching things a bit.

My immediate thought is to be conservative, as we can always white list further files in the future, if this list does open open up even more.

Thoughts @georgestephanis ?

**Comment by jeherve** [comment]
Fixed in 5018944ce8460ed30b5331a40095029937b14f2a
